### PR TITLE
Feat auth

### LINKED ISSUE #6: feat-auth

### PR DESCRIPTION
**Pull request by coldpigli**
What's new:
1) Added protected routes
2) Login Feat
3) Signup Feat
4) Toasts

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -37,30 +37,84 @@ ul{
...
+    background-color: var(--font-dark);
+    color: var(--font-secondary);
+    border: none;
+}
+
+.light-fill{
+    background-color: var(--font-secondary);
+    color: var(--font-dark);
+    border: none;
+}
```
Comment (srejitk) [inline src/App.css:59]: Format the documents once to improve readability

File: src/App.css [hunk 2]
```diff
@@ -37,30 +37,84 @@ ul{
...
+
+.btn-primary-outline{
+    color: inherit;
+    border-color: inherit;
+}
+
+.btn-primary-outline:hover{
+    color: inherit;
+    background-color: transparent;
+    border-color: inherit;
```
Comment (srejitk) [inline src/App.css:98]: Liked how you've inherited the colors here, would make adding a dark theme easier.
Good Work!

**Review by srejitk — APPROVED** [review]
Good use of useLocation and also liked how you've handled implementation of dark themes.
Good work

**Review by nakulsharma15 — COMMENTED** [review]
Looks good to me. No changes suggested. I Liked the way you structured the files in their respective folders and the names which you've given to the variables and functions

**Review by nakulsharma15 — APPROVED** [review]

**Pull request merged**